src/components/CompareResults/ResultsMain.tsx

```diff
@@ -41,6 +62,33 @@ function ResultsMain() {
     }),
   };
 
+  type SearchParamsType = {
+    [key: string]:
+      | string
+      | string[]
+      | Framework['id']
+      | Framework['name']
+      | Repository['name'];
+  };
+
+  const onFrameworkChange = (event: SelectChangeEvent) => {
+    const id = +event.target.value as Framework['id'];
+    setFrameworkIdVal(id);
+    const searchParamsObject = {} as SearchParamsType;
+    searchParams.forEach((_, key) => {
+      const values = searchParams.getAll(key);
+      // account for revs and repos being arrays
```

Comment:
But you're keeping just the first rev/repo with this code, right?
(anyway see below for another implementation)

Reply:
Yes, this is irrelevant with the fix below.